### PR TITLE
new plugin WatchTogetherAdblock: block ads in youtube activity

### PR DESCRIPTION
**Pull request by ImLvna**
Finally got around to pr'ing this

### CODE REVIEW

**Comment by kumpz** [comment]
Thanks for keeping it up to date! 😃

**Comment by CodeF53** [comment]
Does this block ads for everyone without the plugin, or just you.

If its just blocking for you, what do you see during in-video ads?

**Comment by kumpz** [comment]
> Does this block ads for everyone without the plugin, or just you.

Just you.

> If its just blocking for you, what do you see during in-video ads?

This only blocks direct YouTube ads. In-video ads by the creators themselves are not blocked.

**Comment by CodeF53** [comment]
> > If its just blocking for you, what do you see during in-video ads?
> 
> This only blocks direct YouTube ads. In-video ads by the creators themselves are not blocked.

No I meant while everyone is watching a vanilla youtube ad, the ones with the skip button, what do you see.

**Comment by ImLvna** [comment]
It plays normally. Normal watchtogether without this plugin randomizes who gets ads, and if not everyone gets one it just keeps playing, youd have to rewind so the people who got ads wont miss content

**Comment by n1ckoates** [comment]
This plugin works great for me, is there anything else that needs to be done before this can be merged? Would be happy to help.

**Comment by CodeF53** [comment]
> is there anything else that needs to be done before this can be merged? Would be happy to help.

~~Blood sacrifice~~. It just takes a long time to get plugins merged. Stay patient and it will happen eventually.

**Comment by solisinvictum** [comment]
Is there a Statusupdate then or if this Plugin get merged?

Have the same Problem with Friends. Or exist a way to install this plugin byself?

Thanks.

File: src/plugins/watchTogetherAdblock.desktop/index.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+import { Devs } from "@utils/constants";
+import definePlugin from "@utils/types";
+
+// The entire code of this plugin can be found in native.ts
+export default definePlugin({
+    name: "WatchTogetherAdblock",
+    description: "Injects AdGuard to Watch Together",
```
Comment (Vendicated) [inline src/plugins/watchTogetherAdblock.desktop/index.ts:13]:
```suggestion
    description: "Block ads in the WatchTogether activity via AdGuard",
```

File: src/plugins/watchTogetherAdblock.desktop/native.ts [hunk 2]
```diff
@@ -0,0 +1,269 @@
+/*
+ * Vencord, a Discord client mod
+ * Copyright (c) 2023 Vendicated and contributors
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+import { app } from "electron";
+import { getSettings } from "main/ipcMain";
+
+const entrypoint = () => {
```
Comment (Vendicated) [inline src/plugins/watchTogetherAdblock.desktop/native.ts:10]: how often does this script get updated?

File: src/plugins/watchTogetherAdblock.desktop/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# WatchTogetherAdblock
+
+Injects a modifified version of [Adguard's BlockYoutubeAdsShortcut](https://github.com/AdguardTeam/BlockYouTubeAdsShortcut) into the watch together activity
```
Comment (Vendicated) [inline src/plugins/watchTogetherAdblock.desktop/README.md:3]:
```suggestion
Blocks ads in the WatchTogether activity.

Note that this only works for yourself, other users in the activity will still see ads.

Powered by a modifified version of [Adguard's BlockYoutubeAdsShortcut](https://github.com/AdguardTeam/BlockYouTubeAdsShortcut)
```

Comment (pointydev) [reply src/plugins/watchTogetherAdblock.desktop/README.md:3]: Be sure to spell `modified` correctly when updating the description. d:

**Comment by Vendicated** [comment]
looks like watch together frames now use `https://880218394199220334.discordsays.com/` so i have updated it to use that. i also extracted the adguard js into its own file for better maintainability 

unfortunately it seems like the adguard script broke or something, there's an error and the player doesn't load
```
Uncaught (in promise) FirebaseError: Firebase: Error (auth/network-request-failed).
```

**Comment by Vendicated** [comment]
well i fixed the error but i'm still seeing ads :(

maybe there's a more updated script

**Comment by Vendicated** [comment]
okay i'm stupid it does use youtube.com afterall, however it seems to be www.youtube.com which the original script didn't account for

looks like it works now?

**Comment by Vendicated** [comment]
confirmed working 🎉

**Comment by Vendicated** [comment]
thank you luna!!

**Comment by ImLvna** [comment]
thank youu >w<

**Comment by D3SOX** [comment]
Is this supposed to work on Vesktop?

image
image

**Comment by Vendicated** [comment]
yes. make sure you fully close and restart discord

**Comment by sgreenden** [comment]
I don't know if this is the right place to post this, but the plugin was working fine till about a few months ago, where it stopped working, this is consistent across different devices. I enable, disable, re-enable and quit out, restart the computer etc, but no matter what, it just doesn't work. Is this an issue only I am having?

**Pull request merged**